web/src/components/Trace/Trace.tsx

```diff
@@ -1,208 +1,50 @@
+  useUpdateFirstLoadEffect(query.data, test, setIsFirstLoad, dispatch, testResultId, testId);
   return (
-    <main>
-      <Grid>
-        <ReflexContainer style={{height: '100vh'}} orientation="horizontal">
-          <ReflexElement flex={0.6}>
-            <ReflexContainer orientation="vertical">
-              <ReflexElement
-                flex={0.5}
-                className="left-pane"
-                data-tour={GuidedTourService.getStep(GuidedTours.Trace, Steps.Diagram)}
-              >
-                <div className="pane-content">
-                  <Diagram
-                    type={SupportedDiagrams.DAG}
-                    trace={testResultDetails?.trace!}
-                    onSelectSpan={handleOnSpanSelected}
-                    selectedSpan={selectedSpan}
-                  />
-                </div>
-              </ReflexElement>
-              <ReflexElement flex={0.5} className="right-pane">
-                <div className="pane-content" style={{padding: '14px 24px', overflow: 'hidden'}}>
-                  <S.TraceTabs onChange={activeTab => onChangeTab(activeTab)}>
-                    <Tabs.TabPane
-                      tab={
-                        <span data-tour={GuidedTourService.getStep(GuidedTours.Trace, Steps.SpanDetail)}>
-                          Span Detail
-                        </span>
-                      }
-                      key="span-detail"
-                    >
-                      <SpanDetail
-                        resultId={testResultDetails?.resultId}
-                        testId={test?.testId}
-                        span={selectedSpan}
-                      />
-                    </Tabs.TabPane>
-                    <Tabs.TabPane
-                      tab={
-                        <span data-tour={GuidedTourService.getStep(GuidedTours.Trace, Steps.TestResults)}>
-                          Test Results
-                        </span>
-                      }
-                      key="test-results"
-                    >
-                      <TestResults
-                        onSpanSelected={handleOnSpanSelected}
-                        trace={testResultDetails?.trace}
-                        resultId={testResultId}
-                      />
-                    </Tabs.TabPane>
-                  </S.TraceTabs>
-                </div>
-              </ReflexElement>
-            </ReflexContainer>
-          </ReflexElement>
-          <ReflexSplitter />
-          <ReflexElement>
-            <div className="pane-content">
-              <TraceTimeline
-                trace={testResultDetails?.trace!}
-                onSelectSpan={handleOnSpanSelected}
-                selectedSpan={selectedSpan}
-              />
-            </div>
-          </ReflexElement>
-        </ReflexContainer>
-      </Grid>
-    </main>
+    <>
+      <TraceError onRunTest={onRunTest} testId={testId} displayError={displayError} onDismissTrace={onDismissTrace} />
```

Comment:
I think we should avoid rendering both at the same time, we are instantiating both components when only one would be displayed at a time.